visual-js/visual-playwright/src/api.ts

```diff
@@ -318,12 +329,64 @@ ${e instanceof Error ? e.message : JSON.stringify(e)}
       diffingMethod,
     });
 
-    await this.api.createSnapshot({
+    const { diffs } = await this.api.createSnapshot({
       ...meta,
       testName,
       suiteName,
       uploadUuid: uploadId,
     });
+
+    if (!this.uploadedDiffIds[testId]) this.uploadedDiffIds[testId] = [];
+
+    this.uploadedDiffIds[testId].push(...diffs.nodes.map((diff) => diff.id));
+  }
+
+  public async visualResults({ testId }: { testId: string }) {
+    const initialStatusSummary: Record<DiffStatus, number> = {
+      [DiffStatus.Approved]: 0,
+      [DiffStatus.Equal]: 0,
+      [DiffStatus.Unapproved]: 0,
+      [DiffStatus.Rejected]: 0,
+      [DiffStatus.Queued]: 0,
+      [DiffStatus.Errored]: 0,
+    };
+    const { buildId } = getOpts();
+    if (!buildId) {
+      throw new Error(
+        'No Sauce Visual build present, cannot determine visual results.',
+      );
+    }
+
+    // Bypass all API requests if we have no uploaded diff IDs for this test.
+    if (
+      !this.uploadedDiffIds[testId] ||
+      this.uploadedDiffIds[testId].length === 0
+    ) {
+      return initialStatusSummary;
+    }
+
+    return backOff(
+      async () => {
+        const summary = { ...initialStatusSummary };
+        const diffsForTestResult = await this.api.diffsForTestResult(buildId);
+        if (!diffsForTestResult) {
+          return summary;
+        }
+        const filterDiffsById = (diff: { id: string; status: DiffStatus }) =>
+          this.uploadedDiffIds[testId].includes(diff.id);
+
+        return diffsForTestResult.nodes
+          .filter(filterDiffsById)
+          .reduce((statusSummary, diff) => {
+            statusSummary[diff.status]++;
+            return statusSummary;
+          }, summary);
+      },
+      {
+        maxDelay: 10000,
+        numOfAttempts: 10,
```

Comment:
how did we pick those numbers?

Reply:
By rolling a die.

They're just general numbers to not spend too long on a single check, but allow retries. I think in the future it may be good to unify all retry values into a single options object so we can stick with something predefined.